App-Service-Environment-AzFirewall/nestedtemplates/vnet.gov.json

```diff
@@ -0,0 +1,666 @@
+{
```

Comment:
we need to combine these two templates into a single template - use parameters to set values for the route tables, use a copy loop for the route table definition.

Reply:
Here's the problem with this (I have tried to solve this via parameters already and cannot find a more elegant solution):  The default route in the route table is the private IP of the Azure Firewall.  The Azure Firewall IP is derived from the subnet range of the AzureFirewallSubnet, which is itself another parameter.  I'm not sure how to make this dynamic based upon another parameter.  Any suggestions appreciated:

{
                        "name": "DefaultRoute",
                        "properties": {
                            "addressPrefix": "0.0.0.0/0",
                            "nextHopType": "VirtualAppliance",
                            "nextHopIpAddress": "[variables('azureFireWallPrivateIp')]"
                        }
                    }

Reply:
Not sure I follow - you already have that as a variable (derived from a parameter)

Reply:
In the VNet.json, all of the route table entries are for Azure commercial IPs.  In the VNet.gov.json, the routes are for Azure Government IPs.  Normally what I'd do is inject all of those routes as a parameters object EXCEPT I cannot add that DefaultRoute because it relies on the value derived in the variable.

So my options as I see them would be to 1) deploy a super set of the routes for Commercial and Azure Gov (messy if the customer is not in Azure Gov or vice versa) 2) have the user know and understand how to derive the defaultRoute next hop IP address (which would probably get missed much of the time) or 3) have a gov specific template (more maintenance).

I recognize that this is not the most elegant solution, but I'm not sure how else I'd handle it, save drop support for Azure government, which was the impetus for doing most of this.  Again, any suggestions welcome.

Reply:
Are you talking about this: https://github.com/Azure/azure-quickstart-templates/pull/5980/files/9889e0547b114888663157fb89b997cfed23a247#diff-dd9ac5fd67628013ee9b9aa3c197b40cR466 - from what I can tell, you already have that authored correctly using the variables which use the parameters you pass in.  If that still doesn't make sense then let's just start working on it and should be easier to debug if something is actually failing.

Reply:
You'll do something like this:

- pass in an array of address prefixes for the specific cloud
- the first element in the array will be for the default route (0.0.0.0/0)
- create a copy loop to create all the routes
- use the if statement for when copyIndex==0 to add the extra parameters for the defaultRoute, e.g.
  "nextHopType": "[if(equals(copyIndex(), 0), "VirtualAppliance", "Internet")]",
  "nextHopIpAddress": "[if(equals(copyIndex(), 0), variables('azureFireWallPrivateIp'), json('null')]"

I think that's it...

Reply:
I think I get it and will try changing it - thanks.